controllers/ccruntime_controller.go

```diff
@@ -629,15 +629,17 @@ func (r *CcRuntimeReconciler) processDaemonset(operation DaemonOperation) *appsv
 	}
 
 	var runtimeClassNames []string
-	var snapshotter = ""
+	var snapshotter_handler_mapping = ""
 	for _, runtimeClass := range r.ccRuntime.Spec.Config.RuntimeClasses {
 		runtimeClassNames = append(runtimeClassNames, runtimeClass.Name)
-		// FIXME: This will need to be changed by the moment the kata-containers
-		//        payload script supports setting one snapshotter per runtime handler.
-		//        For now, for the v0.8.0 release, we're fine assuming that all the
-		//        set snapshotters are going to be the same.
-		if snapshotter == "" && runtimeClass.Snapshotter != "" {
-			snapshotter = runtimeClass.Snapshotter
+
+		if runtimeClass.Snapshotter != "" {
+			shim := strings.TrimPrefix(runtimeClass.Name, "kata-")
+			if snapshotter_handler_mapping == "" {
+				snapshotter_handler_mapping += shim + ":" + runtimeClass.Snapshotter
+			} else {
+				snapshotter_handler_mapping += "," + shim + ":" + runtimeClass.Snapshotter
+			}
```

Comment:
this `kata-` prefix processing is repeated below. Can we do it only once? I also think preparing `snapshotter_handler_mapping` can follow the same `strings.Join()` approach used by `runtimeClassNames`.

Reply:
So, the prefix processing is done for a different reason here, and looking at the code is quite hard to make this only once, as kata-containers explicitly requires the shim name for all the operations, and for the sake of the user's experience the operator requires a full runtime class name (as in kata-qemu).

About the string.Join(), this can be done, I just need to either make sure `, ` is correctly treated on the kata-containers side, or that we remove the last `,` that would be added by the strings.Join().

Reply:
Cool, I've updated the PR and simplified a little bit the logic.
Thanks for the rveiew, @mythi!